src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java

```diff
@@ -124,4 +147,119 @@ public void removePaymentAccountAndUserAccountMaps(EntityManager em, List<UserAc
+
+        //Update valid PBAs, returns any remaining invalid PBAs
+        if (isNotEmpty(pbasFromRequest)) {
+            invalidPbaResponses = acceptOrRejectPbas(pbasFromRequest, pbaRequestList, invalidPbaResponses);
+        }
+
+        return generateUpdatePbaResponse(invalidPbaResponses, pbasFromRequest);
+    }
+
+    private List<PbaUpdateStatusResponse> getInvalidPbaRequests(
+            List<PbaRequest> pbaRequestList, Set<String> pbasFromRequest, String orgId) {
+
+        List<PbaUpdateStatusResponse> invalidPbaRequests = new ArrayList<>();
+
+        pbaRequestList.forEach(pba -> {
+            //check pba is present
+            if (isNullOrEmpty(pba.getPbaNumber())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse("", ERROR_MSG_PBA_MISSING));
+                //check status is present
+            } else if (isNullOrEmpty(pba.getStatus())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_MISSING));
+                pbasFromRequest.remove(pba.getPbaNumber());
+                //check status is valid
+            } else if (!ACCEPTED.name().equalsIgnoreCase(pba.getStatus())
+                    && !REJECTED.name().equalsIgnoreCase(pba.getStatus())) {
+                invalidPbaRequests.add(new PbaUpdateStatusResponse(pba.getPbaNumber(), ERROR_MSG_STATUS_INVALID));
+                pbasFromRequest.remove(pba.getPbaNumber());
+            }
+        });
+
+        //Check individual PBA numbers are valid and exist within given organisation
+        invalidPbaRequests.addAll(getInvalidPbas(pbasFromRequest, orgId));
+
+        return invalidPbaRequests;
+    }
+
+    public List<PbaUpdateStatusResponse> getInvalidPbas(Set<String> pbasFromRequest, String orgId) {
+        List<PbaUpdateStatusResponse> pbaUpdateStatusResponses = new ArrayList<>();
+
+        pbasFromRequest.forEach(pba -> {
+            //check PBA has a valid format
+            if (isPbaInvalid(pba)) {
+                pbaUpdateStatusResponses.add(new PbaUpdateStatusResponse(pba, ERROR_MSG_PBA_INVALID_FORMAT));
+            } else {
+                Optional<PaymentAccount> paymentAccount = paymentAccountRepository.findByPbaNumber(pba);
+
+                //check PBA exists in DB
+                if (paymentAccount.isPresent()) {
+                    //check PBA belongs to the given Organisation
+                    if (!paymentAccount.get().getOrganisation().getOrganisationIdentifier().equals(orgId)) {
+                        pbaUpdateStatusResponses.add(new PbaUpdateStatusResponse(pba, ERROR_MSG_PBA_NOT_IN_ORG));
+                    }
+                } else {
+                    pbaUpdateStatusResponses.add(new PbaUpdateStatusResponse(pba, ERROR_MSG_PBA_NOT_IN_ORG));
+                }
+            }
+        });
+
+        return pbaUpdateStatusResponses;
+    }
+
+    public List<PbaUpdateStatusResponse> acceptOrRejectPbas(

```

Comment:
This method needs to be revamped to use findByPbaNumber , save, delete db operations only once. This will create performance issues.

Reply:
Addressed here: https://github.com/hmcts/rd-professional-api/pull/1096/commits/db27d8f70138e98e2f119e5d6c7f0f350cba04fa